From 4bf8fea403966b2b7c60a8b87cc443888cf27161 Mon Sep 17 00:00:00 2001
From: Robert Bogart <Robert.bogart@joyent.com>
Date: Mon, 17 Dec 2018 23:39:30 +0000
Subject: [PATCH] MANTA-4047 Follow on to MANTA-3664: Improve documentation in
 README.md Reviewed by: Khang Nguyen <khang.nguyen@joyent.com> Approved by:
 Trent Mick <trentm@gmail.com>

---
 README.md | 72 +++++++++++++++++++++++++++++++++++++++++++++++++++++--
 1 file changed, 70 insertions(+), 2 deletions(-)

diff --git a/README.md b/README.md
index b3ad8d7..52425b4 100644
--- a/README.md
+++ b/README.md
@@ -99,9 +99,9 @@ disk.
 
 After all mako summaries in a region have been obtained (whether downloaded or
 derived from the full mako manifests) and the regional report `region.json` has
-been completed, it is then uploaded to `/poseidon/mako/summary`.
+been completed, it is then uploaded to `/poseidon/stor/mako/summary`.
 
-# Dependencies
+# Requirements
 The automation in `bin` requires that the following are installed:
 * GNU awk 4.1.3 or later: (earlier versions do not allow you to specify the
 level of precision which would limit the user to 53 bits).
@@ -114,6 +114,74 @@ access Manta via the client tools.  More information regarding configuring the
 Manta Client Tools can be found on the
 [Manta Client Tools and SDK](https://github.com/joyent/node-manta) project page.
 
+# Installation Steps
+1. Install nodes.js:
+```
+# pkgin install nodejs-6.14.1
+```
+**Note:** This may work with other versions of node, but 6.14.1 was the version
+that was tested with.
+
+2. Install and configure Manta Command Line Tools:
+```
+# npm install -g manta
+```
+
+If your system does not have an ssh key, one can be generated with the
+following:
+```
+# ssh-keygen -t rsa -b 2048
+```
+
+In order to use the command line tools, it is necessary to set at least the
+following environment variables the first time:
+
+```
+$ export MANTA_KEY_ID=$(ssh-keygen -l -f ~/.ssh/id_rsa.pub | awk '{print $2}')
+$ export MANTA_URL=https://us-east.manta.joyent.com
+$ export MANTA_USER=rbogart
+```
+
+**Note:** 
+* `MANTA_URL` is set based on the desired region.
+* In the example above, the account has operator access and the public key was
+added through the admin portal at https://my.samsungcloud.io.  The alternative
+is to ask to have your public key included in the poseidon account in which
+case, do not forget to set `MANTA_USER` to `poseidon`.
+
+3. Install GNU awk 4.1.3 (or later):
+```
+# pkgin install gawk-4.1.3
+```
+
+4. Install git:
+```
+# pkgin install git
+```
+
+5. Clone the mako-regional-report GitHub repository and run the tool:
+```
+# git clone https://github.com/joyent/mako-regional-report.git
+# cd mako-regional-report
+# ./bin/report.sh
+```
+
+**Note:** It is possible that when generating the report, there are no
+pre-existing summaries for the mako manifests in `/poseidon/stor/mako/summary`.
+As discussed above, the utility will then resort to generating its own summary
+for the mako by deriving it from the full manifest in `/posiedon/stor/mako`.
+This is mentioned again because in the case where the summary is not present
+you will see a message like this appear in the terminal:
+```
+mget: ResourceNotFoundError: /poseidon/stor/mako/summary/1.stor.us-east.scloud.host was not found
+Mon Dec 17 21:31:30 UTC 2018: report.sh: info: Unable to find summary for 1.stor.us-east.scloud.host.
+```
+This is not a sign of anything that has gone wrong -- it simply means that the
+tool will be generating the summary itself since the mako apparently has not
+been configured to do it.  For more information on how to configure a mako to
+generate its own summary, please see the section (below) on enabling
+post-processing of the manifest on the mako.
+
 # Configuration
 In the event that the process of aggregating a region does not complete in a
 timely fasion due to having to derive summaries for all mako manifests, it is
-- 
2.21.0

